### PR TITLE
deps+mobx: upgrade mobx to v6

### PR DESCRIPTION
**Pull request by jamaljsr**
This is just a maintenance update to upgrade the `mobx` dependency to v6.0.2 which cleans up the code a bit.

### CODE REVIEW

File: app/src/store/provider.tsx [hunk 1]
```diff
@@ -18,8 +17,8 @@ const StoreContext = React.createContext<Store | undefined>(undefined);
 export const StoreProvider: React.FC<{
   store: Store;
 }> = ({ children, store }) => {
-  const localStore = useLocalStore(() => store);
-  return <StoreContext.Provider value={localStore}>{children}</StoreContext.Provider>;
+  // const localStore = useLocalStore(() => store);
```
Comment (guggero) [inline app/src/store/provider.tsx:20]: nit: remove commented code?

**Review by guggero — APPROVED** [review]
Can't judge most of the changes as I'm not that deep into TS.
But looks sane from afar, so LGTM :100:

File: app/src/store/provider.tsx (hunk 1 above)
Comment (jamaljsr) [reply app/src/store/provider.tsx]: Grrrr. Not sure how I missed that one. Thanks

Comment (jamaljsr) [reply app/src/store/provider.tsx]: This is fixed now.

**Pull request merged**